src/miral/display_configuration_option.cpp

```diff
@@ -105,15 +110,39 @@ void miral::display_configuration_options(mir::Server& server)
             auto const options = server.get_options();
             auto display_layout = options->get<std::string>(display_config_opt);
             auto with_alpha = options->get<std::string>(display_alpha_opt) == display_alpha_on;
+            auto const scale_str = options->get<std::string>(display_scale_opt);
+
+            double scale{0};
+            try
+            {
+                scale = std::stod(scale_str);
+            }
+            catch (std::invalid_argument const&)
+            {
+            }
+            if (scale < 0.01 || scale > 100.0)
+            {
+                mir::fatal_error("Invalid scale %s", scale_str.c_str());
```

Comment:
It's easy enough here to distinguish between “user passed something that's not a double” and “user passed a value we reject”, and I think it would be useful to tell people that we're going to (perfectly sensibly!) reject scales outside the range 0.01 - 100.0.

Reply:
Done, I'll leave the borsing to you